src/analysis/convex/basic.lean

```diff
@@ -599,6 +599,37 @@ begin
   convert this; symmetry; simp only [div_eq_iff (ne_of_gt B), y]; ring
 end
 
+lemma convex_on.slope_mono_adjacent {s : set ℝ} {f : ℝ → ℝ} {x y z : ℝ} (hx : x ∈ s) (hz : z ∈ s)
+  (hxy : x < y) (hyz : y < z) (hf : convex_on s f) : (f y - f x) / (y - x) ≤ (f z - f y) / (z - y) :=
+begin
+  have h₁ : 0 < y - x := by linarith,
+  have h₂ : 0 < z - y := by linarith,
+  have h₃ : 0 < z - x := by linarith,
+
```

Comment:
Please remove empty lines. If you want to make a break in the proof, add a one-line comment explaining in English what's happening here.